pkg/workspace/util.go

```diff
@@ -237,3 +211,63 @@ func CompleteWholeS3Config(config *workspace.S3Config) {
 		config.Region = region
 	}
 }
+
+// GetIntFieldFromGenericConfig returns the value of the key in config which should be of type int.
+// If exist but not int, return error. If not exist, return 0, nil.
+func GetIntFieldFromGenericConfig(config workspace.GenericConfig, key string) (int, error) {
+	value, ok := config[key]
+	if !ok {
+		return 0, nil
+	}
+	i, ok := value.(int)
```

Comment:
how about using `int32` here?

Reply:
If return `int32`, there should be a new func named `GetInt32FromGenericConfig` cause the returned type is fixed.